### PR TITLE
Forwarding preference can't change for a track

### LINKED ISSUE #351: Is the scope of Forwarding Preference really Objects?
Draft-02 strongly implies the forwarding preference is a per-object decision. thus the following sequences would be legal:
[sequences are written as (group, object)]

Stream 2: (0, 1), (0, 2) (0, 3) (0, 5) -- type is StreamHeaderGroup
Stream 6: (0, 4) -- type is ObjectStream

or

Stream 2: (0, 0), (1, 0), (2, 0), (3, 0) -- type is StreamHeaderTrack
Stream 6: (1,1) -- type is Object Stream

etc.

If someone can imagine a use case for this, that's great. If not, it would be easier to implement with different assumptions, e.g.

(1) Forwarding preference is a track-level property. Thus the preference could be provided in the SUBSCRIBE_OK and we wouldn't evens strictly need four different object codepoints (though you would want the distinct formats, and might want to retain the codepoints for that reason)

(2) Forwarding preference is group-level -- some groups could be aggregated under StreamHeaderTrack, but others could be under StreamHeaderGroup. If there is a StreamHeaderGroup for a group, there MUST NOT be object-level streams for that group.

Again, I have little sense of the use cases out there, but I think we should explicitly decide what we are doing and make it clear in  the doc.

### PR DESCRIPTION
**Pull request by ianswett**
Fixes #351 

Closes #375

### CODE REVIEW

**Review by suhasHere — APPROVED** [review]

File: draft-ietf-moq-transport.md [hunk 1]
```diff
@@ -899,6 +899,11 @@ NOT be processed by a relay.
 
 ### Object Message Formats
 
+Every Track has a single 'Object Forwarding Preference' and publishers
+MUST NOT mix different forwarding preferences within a single track.
+If a subscriber receives different forward prefernces for a track, it
```
Comment (suhasHere) [inline draft-ietf-moq-transport.md:904]: type on prefernces

**Review by martinduke — APPROVED** [review]
This is an improvement, though I would still like to get rid of all the wasted bits

**Review by vasilvv — APPROVED** [review]

File: draft-ietf-moq-transport.md (hunk 1 above)
Comment (ianswett) [inline draft-ietf-moq-transport.md:904]:
```suggestion
If a subscriber receives different forwarding preferences for a track, it
```

**Comment by kixelated** [comment]
> This is an improvement, though I would still like to get rid of all the wasted bits

They won't be wasted until we have 64 message types valid on unidirectional streams. We could save the message type byte if OBJECT is the only message allowed on unidirectional streams, but that's not forward-compatible.

**Pull request merged**